src/screenlock/screenlock.cpp

```diff
@@ -204,11 +209,36 @@ void ScreenLock::setScreenLocked(bool value)
 {
     // TODO Make the view a lock screen view (title? stacking layer?)
     if (m_lockscreenVisible != value) {
+        if (m_allowSkippingLockScreen) {
+            // Allow disabling lock screen, but only if the device is not locked
+            if (value && !m_deviceIsLocked) {
+                return;
+            }
+        }
         m_lockscreenVisible = value;
         emit screenLockedChanged(value);
     }
 }
 
+bool ScreenLock::isDeviceLocked() const
+{
+    return m_deviceIsLocked;
+}
+
+void ScreenLock::setDeviceIsLocked(bool locked)
+{
+    if (m_deviceIsLocked != locked) {
+        m_deviceIsLocked = locked;
+        emit deviceIsLockedChanged(locked);
+
+        if (m_allowSkippingLockScreen) {
+            if (locked && displayState() == TouchScreen::DisplayOff) {
+                lockScreen();
```

Comment:
Is this display off special case really needed?
I have a feeling that it might interfere with mce side autolock corner cases.